src/rules-parser.pegjs

```diff
@@ -83,7 +107,7 @@ Schema =
       result.properties = properties.properties;
     }
     if (derived) {
-      result.derivedFrom = derived[2];
+      result.derivedFrom = ensureUpperCase(derived[2], "Type names");
```

Comment:
you will get two warnings, one from declaring the type name with uppercase, and then for every use of it within an extends